lib/cloud_controller/opi/apps_client.rb

```diff
@@ -104,10 +104,35 @@ def to_hash
       end
     end
 
+    class KpackLifecycle
+      CNB_LAUNCHER_PATH = '/cnb/lifecycle/launcher'.freeze
+
+      def initialize(process)
+        @process = process
+      end
+
+      def to_hash
+        command = if @process.command.presence
+                    ['/bin/sh', '-c', CNB_LAUNCHER_PATH, @process.command]
```

Comment:
I think this command should actually be:
```ruby
['/bin/sh', '-c', "#{CNB_LAUNCHER_PATH} #{@process.command}"]
```

where the launcher and specified command are a single string that's passed to the shell.

The app I had tested this with was working, but I think it may have been running the images default entrypoint? I don't know... a bit confused by that. Anyways going to add a commit for this.

Reply:
https://github.com/cloudfoundry/cloud_controller_ng/pull/1917/commits/6a32abbd93ca77604f2657347cdb6522b647f857